### DISCUSSION TITLE
Building DiligentCore MSVC

### DISCUSSION THREAD

**Discussion by AnloWho** [DiligentGraphics/DiligentCore#624] [Q&A] [answered]
Hi, I'm exploring the possibilities to use Dilgent Engine for a project. However, I only want the minimum. Is compiling DilgentCore enough to get started? I face #inlucde issues for std::min and std::max here and there. Is that normal?
I'm on Windows 10, compiling with MSVC v141 toolchain.

**Comment by TheMostDiligent** [comment]
DilgentCore proivdes all graphics backend and everything needed to start building cross-platform graphics application.
For min/max, can you point out where the issue happens? You can define `NOMINMAX` before including windows headers, or at the project level, but the headers should compile without it too.

**Reply by AnloWho** [reply]
Diligent-GraphicsTools/BufferSuballocator.h:110 
Diligent-Common/ThreadPool.cpp:122
Diligent-GraphicsEngine/AsyncInitializer.hpp:79

makes use of std::max or std::min but I don't see any ```#include <algorithm>``` anywhere 
Maybe some indirect includes somewhere, but I wouldn't rely on that. I know this can be different on gcc, but msvc seems picky.
I'm trying to build DiligentCore-static only. As said, toolchaing v141 with WinSDK 10.0.22621.0. Debug build.

**Reply by TheMostDiligent** [reply, marked as answer]
Try this: https://github.com/DiligentGraphics/DiligentCore/commit/12b06b82506d32f8de300e9537375f893b723a57

**Reply by AnloWho** [reply]
Yes, that makes the compilation run through. Now I'm facing some linking issues, I will look into it.
